File: server/src/main/java/bgu/spl/net/impl/stomp/Service/STOMP_Frames/SendFrame.java
package bgu.spl.net.impl.stomp.Service.STOMP_Frames;

import java.io.IOException;
import java.util.HashMap;

import bgu.spl.net.genericServers.interfaces.ConnectionHandler;

public class SendFrame extends ExecutableFrame {

    private static final Object DESTINATION_FIELD = "destination";

    protected SendFrame(HashMap<String,String> headers, String frameBody) {
        super(headers, frameBody, StompCommand.SEND);
    }

    @Override
    public String execute(ConnectionHandler<String> handler) {
        
        try {
            channelsManager.broadcast(headers.get(DESTINATION_FIELD), frameBody);
            return ""; //this frame has no response - not sure if this will work
        } catch (IOException e) {
            return ErrorFrame.get(e.getMessage()).toString();
        }
    }
}

File: client/src/frames/SendFrame.cpp
#include "SendFrame.h"
#include "event.h"
#include <unordered_map>
#include "UserData.h"

SendFrame::SendFrame(unordered_map<string, string> headers, string frameBody)
    : Frame(StompCommand::SEND, headers, frameBody){}


SendFrame *SendFrame::get(Event& event)
{
    unordered_map<string, string> headers;
    headers["destination"] = "/" + event.get_team_a_name() + "_" + event.get_team_b_name();
    string body = event.extractFrameBody();
    return new SendFrame(headers, body);
}
